NETDefault/Gateway.Buckaroo/BuckarooGateway.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Gateway.Buckaroo.Exceptions;
+using Gateway.Buckaroo.Mappers;
+using Gateway.Buckaroo.Models.Dto.Requests;
+using Gateway.Buckaroo.Models.Dto.Responses;
+using Gateway.Buckaroo.Models.Entities;
+using Gateway.Buckaroo.Models.Request;
+using Microsoft.Extensions.Logging;
+
+namespace Gateway.Buckaroo;
+
+public class BuckarooGateway : ISubscriptionGateway
+{
+    private readonly ILogger<BuckarooGateway> _logger;
+    private readonly IPspClient _client;
+
+    public BuckarooGateway(
+        ILogger<BuckarooGateway> logger,
+        IPspClient client)
+    {
+        _logger = logger;
+        _client = client;
+    }
+
+    public async Task<Debtor?> GetDebtorAsync(GetDebtorRequest requestModel)
+    {
+        var buckarooRequest = BuckarooRequestMapper.Map(requestModel);
+
+        try
+        {
+            var buckarooResponse = await _client.PostDataRequestAsync<GetDebtorInfoRequestDto, DebtorInfoResponseDto>(buckarooRequest);
+
+            if (buckarooResponse!.Status!.Code!.Description!.Contains("Success"))

```

Comment:
I don't like the look of these bangs. If those properties are nullable, it can be null sometimes. Either check, ore change the signature of those properties.